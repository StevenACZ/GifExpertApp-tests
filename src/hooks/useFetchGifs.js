import { useEffect, useState } from "react"
import getGifsFetch from "../helpers/getGifsFetch";

export const useFetchGifs = ( category ) => {
  const [state, setState] = useState({
    data: [],
    loading: true
  });

  useEffect( () => {
    getGifsFetch( category )
      .then( imgs => setState({
        data: imgs,
        loading: false
      }));
  }, [ category ]);

  return state; // { data: [], loading: true };
}