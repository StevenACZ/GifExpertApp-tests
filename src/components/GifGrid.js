import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import GifGridItem from './GifGridItem';

const GifGrid = ( { category } ) => {
  const { data:images, loading } = useFetchGifs( category );

  return (
    <div>
      <h3>{ category }</h3>

      { loading && <p>Cargando...</p> }

      {
        images.map( img => (
          <GifGridItem
            key={ img.id }
            { ...img }
          />
        ))
      }
    </div>
  )
};

export default GifGrid;