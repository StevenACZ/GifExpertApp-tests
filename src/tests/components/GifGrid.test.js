import { shallow } from "enzyme";
import GifGrid from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en <GifGrid />', () => {
  const category = 'One punch';
  
  test('debe de mostrarse el componente', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true
    });
    
    const wrapper = shallow( <GifGrid category={ category } /> );
    expect( wrapper ).toMatchSnapshot();
  });
  
  test('debe de mostrar items cuando se cargan imagenes useFetchGifs', () => {
    const gifs = [
      {
        id: 'ABC',
        url: 'https://localhost/dsd/cosa.jpg',
        title: 'Cualquier cosa'
      },
      {
        id: 'ASDSAD',
        url: 'https:///localhost/dsd/cosa.jpg',
        title: 'cosa'
      }
    ] 
    
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false
    });
    
    const wrapper = shallow( <GifGrid category={ category } /> );
    expect( wrapper.find('p').exists() ).toBe( false );
    expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );
  })
});