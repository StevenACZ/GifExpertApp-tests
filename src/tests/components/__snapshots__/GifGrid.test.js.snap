// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> debe de mostrar items cuando se cargan imagenes useFetchGifs 1`] = `
<Fragment>
  <h3>
    One punch
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="https://localhost/dsd/cosa.jpg"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en <GifGrid /> debe de mostrarse el componente 1`] = `
<Fragment>
  <h3>
    One punch
  </h3>
  <p>
    Cargando...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
